core/component-controller/src/main/kotlin/com/merxury/blocker/core/controllers/shizuku/ShizukuAppController.kt

```diff
@@ -0,0 +1,249 @@
+            pm?.setApplicationEnabledSetting(
+                packageName,
+                COMPONENT_ENABLED_STATE_ENABLED,
+                0,
+                userId,
+                context.packageName,
+            )
+        } else {
+            pm?.setApplicationEnabledSetting(
+                packageName,
+                COMPONENT_ENABLED_STATE_ENABLED,
+                0,
+                userId,
+            )
+        }
+        return true
+    }
+
+    override suspend fun clearCache(packageName: String): Boolean {
+        Timber.i("Start clear cache: $packageName")
+        val userId = Shizuku.getUid()
+        if (userId == SHELL_UID) {
+            openAppDetails(packageName)
+            return true
+        }
+        return suspendCoroutine { cont ->
+            pm?.deleteApplicationCacheFiles(
+                packageName,
+                object : IPackageDataObserver.Stub() {
+                    override fun onRemoveCompleted(packageName: String?, succeeded: Boolean) {
+                        Timber.i("Clear cache for $packageName succeeded: $succeeded")
+                        cont.resumeWith(Result.success(succeeded))
+                    }
+                },
+            )
+        }
+    }
+
+    override suspend fun clearData(packageName: String): Boolean {
+        Timber.i("Start clear data: $packageName")
+        val userId = Shizuku.getUid()
+        if (userId == SHELL_UID) {
+            openAppDetails(packageName)
+            return true
+        }
+        return suspendCoroutine { cont ->
+            pm?.clearApplicationUserData(
+                packageName,
+                object : IPackageDataObserver.Stub() {
+                    override fun onRemoveCompleted(packageName: String?, succeeded: Boolean) {
+                        Timber.i("Clear data for $packageName succeeded: $succeeded")
+                        cont.resumeWith(Result.success(succeeded))
+                    }
+                },
+                userId,
+            )
+        }
+    }
+
+    override suspend fun uninstallApp(packageName: String, versionCode: Long): Boolean {
+        val broadcastIntent = Intent("com.merxury.blocker.UNINSTALL_APP_RESULT_ACTION")
+        val intent = PendingIntent.getBroadcast(
+            context,
+            0,
+            broadcastIntent,
+            PendingIntent.FLAG_UPDATE_CURRENT or PendingIntent.FLAG_IMMUTABLE,
+        )
+        val isSystemApp = ApplicationUtil.isSystemApp(context.packageManager, packageName)
+        // 0x00000004 = PackageManager.DELETE_SYSTEM_APP
+        // 0x00000002 = PackageManager.DELETE_ALL_USERS
+        val flags = if (isSystemApp) 0x00000004 else 0x00000002
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P) {
+            packageInstaller?.uninstall(
+                VersionedPackage(packageName, versionCode),
+                context.packageName,
+                flags,
+                intent.intentSender,
+                0,
```

Comment:
@SanmerDev Did you suggested using `UserHandle.getUserId(int)` in here?

Reply:
I'm reviewing it, just leave it to me.